server/api/launcher.py

```diff
@@ -224,6 +225,7 @@ def _pre_run_enrichement(self, runtime, run):
         This ensures the node selector is correctly set on the run
         while maintaining the runtime's integrity from system-specific project settings.
         """
+        run.spec.node_selector = deepcopy(runtime.spec.node_selector)
```

Comment:
I think we need a better differentiation between this function and `_enrich_run` because that also happens "pre run". So when I saw this line I thought it should be in `_enrich_run` but I guess you want it here because it's related to node selector. I think it should be called `_enrich_run_node_selector`

Reply:
Changed to `_pre_run_node_selector_enrichement`, wdyt?